src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -61,6 +61,48 @@
  * <pre>
  * &lt;module name=&quot;NoWhitespaceBefore&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * int foo;
+ * foo ++; // violation, whitespace before '++' is not allowed
+ * foo++; // OK
+ * for (int i = 0 ; i &lt; 5; i++) {}  // violation
+ *            // ^ whitespace before ';' is not allowed
+ * for (int i = 0; i &lt; 5; i++) {} // OK
+ * int[][] array = { { 1, 2, 3 }
+ *                 , { 4, 5, 6 }// violation
+ *                 , { 7, 8, 9 } }; // violation, whitespace before ',' is not allowed
+ * int[][] array2 = { { 1, 2, 3 },
+ *                    { 4, 5, 6 },
+ *                    { 7, 8, 9 } }; // OK
+ * Customer c = new Customer(); // Customer is a user-defined class
+ * c.setFirstName("John")
+ *  .setLastName("Doe")
+ *  ; // violation, linebreak is not allowed before ';' token
+ * </pre>
+ * <p>To configure the check to allow linebreaks before default tokens:</p>
+ * <pre>
+ * &lt;module name=&quot;NoWhitespaceBefore&quot;&gt;
+ *   &lt;property name=&quot;allowLineBreaks&quot; value=&quot;true&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * int[][] array = { { 1, 2, 3 }
+ *                 , { 4, 5, 6 } // OK
+ *                 , { 7, 8, 9 } }; // OK, linebreak is allowed before default tokens
```

Comment:
`is not allowed before ','`.

It is hard to remember what is default. Please fix all other cases like this.

Reply:
Done